test/autorest/lrogroup/lros_test.go

```diff
@@ -69,1142 +75,974 @@ func TestLROBeginDelete202NoRetry204(t *testing.T) {
+	if err == nil {
+		t.Fatal("expected an error but did not receive one")
+	}
+}
 
-// func TestLROBeginPutAsyncNonResource(t *testing.T) {
-// 	op := getLROSOperations(t)
-// 	poller, err := op.BeginPutAsyncNonResource(context.Background(), nil)
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	rt, err := poller.ResumeToken()
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	poller, err = op.ResumeLrOSPutAsyncNonResourcePoller(rt)
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	for poller.Poll(context.Background()) {
-// 		time.Sleep(200 * time.Millisecond)
-// 	}
-// 	resp, err := poller.Response()
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	helpers.VerifyStatusCode(t, resp.RawResponse, 200)
-// 	resp, err = poller.Wait(context.Background(), time.Duration(1)*time.Second)
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	helpers.VerifyStatusCode(t, resp.RawResponse, 200)
-// }
+func TestLROBeginPutAsyncNonResource(t *testing.T) {
+	op := getLROSOperations(t)
+	resp, err := op.BeginPutAsyncNonResource(context.Background(), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	poller := resp.Poller
+	rt, err := poller.ResumeToken()
+	if err != nil {
+		t.Fatal(err)
+	}
+	poller, err = op.ResumePutAsyncNonResource(rt)
+	if err != nil {
+		t.Fatal(err)
+	}
+	res, err := resp.PollUntilDone(context.Background(), 1*time.Millisecond)
+	if err != nil {
+		t.Fatal(err)
+	}
+	helpers.VerifyStatusCode(t, res.RawResponse, 200)
+	helpers.DeepEqualOrFatal(t, res.Sku, &lrogroup.Sku{
+		ID:   to.StringPtr("100"),
+		Name: to.StringPtr("sku"),
+	})
+}
 
-// func TestLROBeginPutAsyncRetryFailed(t *testing.T) {
-// 	t.Skip("CloudError unmarshalling failed")
-// 	op := getLROSOperations(t)
-// 	poller, err := op.BeginPutAsyncRetryFailed(context.Background(), nil)
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	rt, err := poller.ResumeToken()
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	poller, err = op.ResumeLrOSPutAsyncRetryFailedPoller(rt)
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	for poller.Poll(context.Background()) {
-// 		time.Sleep(200 * time.Millisecond)
-// 	}
-// 	resp, err := poller.Response()
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	helpers.VerifyStatusCode(t, resp.RawResponse, 200)
-// 	resp, err = poller.Wait(context.Background(), time.Duration(1)*time.Second)
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	helpers.VerifyStatusCode(t, resp.RawResponse, 200)
-// }
+func TestLROBeginPutAsyncRetryFailed(t *testing.T) {
+	t.Skip("CloudError unmarshalling failed")
+	op := getLROSOperations(t)
+	resp, err := op.BeginPutAsyncRetryFailed(context.Background(), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	poller := resp.Poller
+	rt, err := poller.ResumeToken()
+	if err != nil {
+		t.Fatal(err)
+	}
+	poller, err = op.ResumePutAsyncRetryFailed(rt)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = resp.PollUntilDone(context.Background(), 1*time.Millisecond)
+	if err == nil {
```

Comment:
Should we be checking that this is a specific error type as defined in the swagger?

Reply:
We should, I'll be adding this when we update the testserver version, since we currently have CloudError code that breaks because of an incorrectly named field.